x-pack/filebeat/input/httpjson/split.go

```diff
@@ -255,7 +255,7 @@ func toMapStr(v interface{}) (common.MapStr, bool) {
 		return t, true
 	case map[string]interface{}:
 		return common.MapStr(t), true
-	case string, []interface{}:
+	case string, []bool, []int, []string, []interface{}:
```

Comment:
Apologies for the slow response.

While this work for the existing tests, I'm not convinced for the general case. What other JSON array types might we see (`[]map[string]interface{}` seems like a possibility)? and what do we do if the `[]interface{}` is an array of unhandleable types (do we care)?

Reply:
Ya it could be infinite 😬. I think though subarrays of maps/arrays may not need to be handled. If they end up wanting to split those it will just recursively handle that...I think. I would say try to cover the basics and if there are changes needed in the future, look at the situation then??

Reply:
Yeah, I think that's fair, though it won't recursively handle it, that would require switching on kind and so on reflect. It might be worth putting a note here that this is where to look when this king of thing happens in the future.

In the mean time it's worth doing a survey (rough) of what types do end up in arrays. I would not be surprised if `float64` might in some contexts for metrics (maybe).

Reply:
@efd6 The original scenario from ticket #30345 includes an example that shows the columns with the corresponding types.  I'm not sure if this helps or makes it more complicated.  Since it parallels how _xpack/sql is working, so being able to consume the types Elastic maps to SQL responses seems like it would cover most common types with fallback to a string.  Since I'm currently reassembling the documents by a making key/value pairs out of the columns per row, actually consuming a source like this natively would be ideal.  Just so we don't get stuck in the weeds of a specific scenario that triggered this.

BTW, in my original source the `columns` and `rows` sections are called something else, so if this is actually what is pursued then having flexibility on those would be important.

Thanks!

Reply:
Thanks @wasserman. We should be able to get away with adding `[]map[string]interface{}` and `[]common.MapStr` to this type list. In order to catch cases that we weren't expecting, I'm wondering if this should not return an error explaining the type it's rejecting rather that a bool, `fmt.Errorf("unexpected type for split: %T", t)` or similar.